.github/scripts/find_gem_version_bounds.rb

```diff
@@ -0,0 +1,140 @@
+    runtime = File.basename(gemfile_name).split('_').first # ruby or jruby
+    File.foreach(gemfile_name) do |line|
+      if (gem_details = parse_gemfile_entry(line))
+        gem_name, version = gem_details
+      elsif (gem_details = parse_gemfile_lock_entry(line))
+        gem_name, version = gem_details
+      else
+        next
+      end
+
+      # Validate and store the minimum version
+      if version_valid?(version)
+        if runtime == 'ruby'
+          if minimum_gems_ruby[gem_name].nil? || Gem::Version.new(version) < Gem::Version.new(minimum_gems_ruby[gem_name])
+            minimum_gems_ruby[gem_name] = version
+          end
+          if maximum_gems_ruby[gem_name].nil? || Gem::Version.new(version) > Gem::Version.new(maximum_gems_ruby[gem_name])
+            maximum_gems_ruby[gem_name] = version
+          end
+        end
+        if runtime == 'jruby'
+          if minimum_gems_jruby[gem_name].nil? || Gem::Version.new(version) < Gem::Version.new(minimum_gems_jruby[gem_name])
+            minimum_gems_jruby[gem_name] = version
+          end
+          if maximum_gems_jruby[gem_name].nil? || Gem::Version.new(version) > Gem::Version.new(maximum_gems_jruby[gem_name])
+            maximum_gems_jruby[gem_name] = version
+          end
+        end
+      else
+        next
+      end
+    end
+  end
+
+  [minimum_gems_ruby, minimum_gems_jruby, maximum_gems_ruby, maximum_gems_jruby]
+end
+
+# Helper: Parse a Gemfile-style gem declaration
+# ex. gem 'ruby-kafka', '~> 5.0'
+def parse_gemfile_entry(line)
+  if (match = line.match(/^\s*gem\s+["']([^"']+)["']\s*,?\s*["']?([^"']*)["']?/))
+    gem_name, version_constraint = match[1], match[2]
+    version = extract_version(version_constraint)
+    [gem_name, version]
+  end
+end
+
+# Helper: Parse a Gemfile.lock-style entry
+# matches on ex. actionmailer (= 6.0.6)
+def parse_gemfile_lock_entry(line)
+  if (match = line.match(/^\s*([a-z0-9_-]+)\s+\(([^)]+)\)/))
+    [match[1], match[2]]
+  end
+end
+
+# Helper: Validate the version format
+def version_valid?(version)
+  version =~ /^\d+(\.\d+)*$/
+end
+
+# Helper: Extract the actual version number from a constraint
+# Matches on the following version patterns:
+# 1. "pessimistic" versions, ex. '~> 1.2.3'
+# 2. '>= 1.2.3'
+# 3. 1.2.3
+def extract_version(constraint)
+  if constraint =~ /~>\s*([\d.]+(?:[-.\w]*))| # Handles ~> constraints
+                     >=\s*([\d.]+(?:[-.\w]*))| # Handles >= constraints
+                     ([\d.]+(?:[-.\w]*))       # Handles plain versions
+                    /x
+    Regexp.last_match(1) || Regexp.last_match(2) || Regexp.last_match(3)
+  end
+end
+
+def get_integration_names(directory = 'lib/datadog/tracing/contrib/')
+  unless Dir.exist?(directory)
+    puts "Directory '#{directory}' not found!"
+    return []
+  end
+
+  # Get all subdirectories inside the specified directory
+  Dir.children(directory).select do |entry|
+    File.directory?(File.join(directory, entry))
+  end
```

Comment:
You need to add this to the top of the file too. It loads the current directory's version of the `datadog` gem to memory:
```ruby
lib = File.expand_path('lib', __dir__)
$LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
require 'datadog'
```

```suggestion
  Datadog::Tracing::Contrib::REGISTRY.map{ |i| i.name.to_s }
```